app/Models/Forum/Topic.php

```diff
@@ -126,25 +129,35 @@ public function addPost($poster, $body)
 
     public function removePost($post, $user = null)
     {
-        DB::transaction(function () use ($post, $user) {
-            $post->delete();
+        $this->validationErrors()->reset();
 
-            if ($this->posts()->exists() === true) {
-                $this->refreshCache();
-            } else {
-                $this->delete();
-            }
+        try {
+            return DB::transaction(function () use ($post, $user) {
+                if ($post->delete() === false) {
+                    $this->validationErrors()->addTranslated('post', $post->validationErrors()->toSentence());
```

Comment:
optionally can use `validationErrors()->merge(..)`

Reply:
but one is errors for post and the other is for topic...